metadata-ingestion/src/datahub/ingestion/transformer/extract_dataset_tags.py

```diff
@@ -0,0 +1,62 @@
+import re
+from enum import Enum
+from typing import List, Optional, cast
+
+from datahub.configuration.common import TransformerSemanticsConfigModel
+from datahub.emitter.mce_builder import Aspect
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.transformer.dataset_transformer import DatasetTagsTransformer
+from datahub.metadata.schema_classes import GlobalTagsClass, TagAssociationClass
+from datahub.utilities.urns.dataset_urn import DatasetUrn
+
+
+class ExtractTagsOption(Enum):
+    URN = "urn"
+
+
+class ExtractDatasetTagsConfig(TransformerSemanticsConfigModel):
+    extract_tags_from: ExtractTagsOption = ExtractTagsOption.URN
+    extract_tags_regex: str
+
+
+class ExtractDatasetTags(DatasetTagsTransformer):
+    """Transformer that add tags to datasets according to configuration by extracting from metadata."""
```

Comment:
imo the naming/docs here are also kinda confusing - this transformer enables you to create tags based on the dataset name, right?

what if we called this `extract_tags_from_dataset_name` or something?

Reply:
Currently it is from the `name` but there is no reason we cannot extract the tags in the future from something else. Want to keep that extensibility for future.

Reply:
ok we can leave the name, but we should at least add some comments here that it _currently_ extracts from the dataset name only

Reply:
Added